py_trees/src/py_trees/blackboard.py

```diff
@@ -105,26 +108,142 @@ def __str__(self):
+    Provides methods to initialize SubBlackboards to watch subset of Blackboard variables
+    And publishers for publishing when variables in Blackboards or SubBlackboards are changed
     """
-    def __init__(self, blackboard):
-        self.blackboard = blackboard
+    class SubBlackboard(object):
+        """
+        Container class for a Subblackboard
+        Keeps track of variables from Blackboard
+        """
+        def __init__(self, topic_name, attrs):
+            self.blackboard = Blackboard()
+            self.topic_name = topic_name
+            self.attrs = attrs
+            self.dict = {}
+            self.cached_dict = {}
+            self.publisher = rospy.Publisher("~" + topic_name, std_msgs.String, latch=True, queue_size=2)
+
+        def update_sub_blackboard(self):
+            for attr in self.attrs:
+                if '/' in attr:
+                    check_attr = operator.attrgetter(".".join(attr.split('/')))
+                else:
+                    check_attr = operator.attrgetter(attr)
+                try:
+                    value = check_attr(self.blackboard)
+                    self.dict[attr] = value
+                except AttributeError:
+                    pass
+
+        def is_changed(self):
+            self.update_sub_blackboard()
+            current_pickle = dumps(self.dict, -1)
+            blackboard_changed = current_pickle != self.cached_dict
+            self.cached_dict = current_pickle
+
+            return blackboard_changed
+
+        def __str__(self):
+            s = "\n"
+            max_length = 0
+            for k in self.dict.keys():
+                max_length = len(k) if len(k) > max_length else max_length
+            keys = sorted(self.dict)
+            for key in keys:
+                value = self.dict[key]
+                if value is None:
+                    value_string = "-"
+                    s += console.cyan + "  " + '{0: <{1}}'.format(key, max_length + 1) + console.reset + ": " + console.yellow + "%s\n" % (value_string) + console.reset
+                else:
+                    lines = ("%s" % value).split('\n')
+                    if len(lines) > 1:
+                        s += console.cyan + "  " + '{0: <{1}}'.format(key, max_length + 1) + console.reset + ":\n"
+                        for line in lines:
+                            s += console.yellow + "    %s\n" % line + console.reset
+                    else:
+                        s += console.cyan + "  " + '{0: <{1}}'.format(key, max_length + 1) + console.reset + ": " + console.yellow + "%s\n" % (value) + console.reset
+            s += console.reset
+            return s
+
+    def __init__(self):
+        self.blackboard = Blackboard()
         self.cached_blackboard_dict = {}
 
-    def is_changed(self):
-        # For simplicity, creating a reference; reference because its a borg
-        blackboard_dict = self.blackboard.__dict__
+        if rospy.core.is_initialized():
```

Comment:
Rather than introduce a different mechanism for detecting when ros comms should be initialised or not, make use of the existing `setup` mechanism to instantiate the ros blackboard along with the other publishers. Then call the parent behaviour tree class's `setup` function.